lib/blocks.php

```diff
@@ -296,3 +296,209 @@ function gutenberg_enqueue_editor_block_styles_assets() {
 	}
 	add_action( 'enqueue_block_editor_assets', 'gutenberg_enqueue_editor_block_styles_assets' );
 }
+
+/**
+ * Renders the classNames and styles for blocks
+ *
+ * @param string $block_content Output of the current block.
+ * @param array  $block Block Object.
+ * @return string New block output.
+ */
+function gutenberg_experimental_apply_classnames_and_styles( $block_content, $block ) {
```

Comment:
what happens with existing blocks that already apply these flags on the frontend on their "save" function? how can we exclude them from this behavior?

Reply:
> what happens with existing blocks that already apply these flags on the frontend on their "save" function?

They go through this function unnecessarily.  They do 'play nice', although this function handling them as well is a bit redundant.

> how can we exclude them from this behavior?

Im not sure.  There doesn't seem to be anything on the block object passed into the function that signifies anything special about this case.  Maybe we can add an extra property on the block object or an attr to signal that it should be ignored by this function?  Im open to ideas...

Reply:
That seems to do the trick!  Thank you 😄  - 7c17c76